### ISSUE TITLE
pkgj compatibility to iTLS-Enso for HTTPS on 3.60?

### ISSUE THREAD

**Issue by Zeitwaechter** [blastrock/pkgj#212]
## General Information

Version of firmware: 3.60 Henkaku-11+Enso v1.1
Version of PKGj: 0.37

## Steps to reproduce the issue

- Run/Setup [iTLS-Enso](https://github.com/SKGleba/iTLS-Enso) for TLSv1.2 Support on 3.60
- Trying to refresh all Sources after amending the http to https

## Obtained result
- Error: 0x80431075 (HTTPS URLs may not be supported on your device)

## Expected result (User Story)
- After patching/loading the compatibility Patch for 3.60 (yay, somewhat-privacy is back), I'm able to refresh/download all Games via HTTPS.

------

Yoho! :)

I don't know the details about how _iTLS-Enso_ "patches" the missing compatibility for TLSv1.2 but in the Browser it works just fine.

My guess is that _pkgj_ could be able to detect [SKGleba](https://github.com/SKGleba) changes and adept to it..?

Thank you very much in advance!

Kind regards :+1:

**Comment by windsurfer1122** [comment]
Same issue here on FW 3.65 + h-encore + Enso.
According to SKGleba iTLS-Enso should work with everything that uses built in tls.

https://github.com works fine in standard Web Browser with iTLS-Enso

**Comment by blastrock** [comment]
I'll look into it. It's strange, pkgj doesn't do anything fancy and just uses the built in HTTP APIs, so it should just work.

**Comment by windsurfer1122** [comment]
I could reproduce this issue on a Fat Vita 3.68 with h-encore: pkgj 0.37 cannot download via https.
url_games https://nopaystation.com/tsv/PSV_GAMES.tsv
My browser is connecting with TLS v1.2 to this URL.
So it seems it's a more general issue with pkgj.

**Comment by Zeitwaechter** [comment]
Only for updates sake:

I just tested out the beta of v0.38 which was just uploaded about 6 hours ago (at the time of writing).

Sadly I'm still getting the same old error even after installing the also new iTLS v2.0 - which was also released very recently.

----

Off Topic:
Btw., I like the new details layer after pressing X on PSV Games @blastrock.

**Comment by Zeitwaechter** [comment]
Hey there!

As I'm not into the whole VitaSDK/Vita Homebrew Development nor any C-thing or VirtualDevEnv/Framework directly above it I sadly won't be of any help.

If I knew what to do I'd already have created a PR.. :(


Therefore as disclaimer: This might not be helping at all - which is what I guess.

I do not have that much Homebrew installed but I found another reference, to how the TLS-Feature could be implemented, within (devnoname120/vhbb)[https://github.com/blastrock/pkgj/blob/4c98fc2a2d97729423a5e5bd4b85057a15153f47/src/vita.cpp#L442].

Maybe those CURL, CURLpp and OpenSSL PKGs which were used within VHBB could also help?
The native CAs should be updated via iTLS' 3.68 CAs so maybe those two factions (PKGs + Certs) could be mixed together to result in a successful handshake?

Greetings! :)


(Big shout out to @devnoname120 / Arkanite for their work on VHBB!)

**Comment by bombsimon** [comment]
Any news for this issue? I’ve been running 3.65 with h-encore and PKGj v0.51 without any problems. Today I had to install iTLS-Enso to get TLS 1.2 support for my hotel wifi and now PKGj stopped working when trying to refresh. The error message I get however is the one mentioned in #306 but since this issue started after installing iTLS I guess it’s related.

**Comment by subworx** [comment]
> **EDIT3**: I have no idea what's going on here but before I was going to dig into this I tried to setup a build environment and built a version from the `master` branch. I re-installed PKGj from the built VPK and suddenly this issue disappears. I then installed the tagged release from GitHub and the error comes back.
> 
> I'm not sure if the latest release is built from `master` and I'm not sure why my build is working but so far so good. If anyone is interested I used Docker to setup the build environment so I'll share it here. I'm installing `ftp` and `curl` since the README states that you should be able to run `make send` with `PSVITAIP` set to your PS Vita IP although I didn't get that to work so I copied the files manually.

Hi @bombsimon , thank you for that info + docker file.
I managed to build it from a freshly cloned master source tree, and got a vpk as result.
After removing the old version and installing the new one, the bug persists.
I re-ran iTLS-Enso, installed the compat module and certificates, and made sure https sites (GitLab) open fine in the browser.
Would you mind posting your compiled version somewhere?

Thanks and best regards,
Sub

**Comment by bombsimon** [comment]
@subworx Hey!

What error message do you get? I'm asking because it's been a while and when i picked up my Vita to respond to you I couldn't refresh the list, got error `0x80436007`.

This is not the original error I had with TLS so I did some research and found [this Reddit post](https://www.reddit.com/r/VitaPiracy/comments/e18tq0/psa_temp_solution_for_pkgj_issue). What that post described made it work again.

I'm not sure this will help you, morer a note to self, but before I found that fix I was trying to confirm PGKj worked without iTLS but I couldn't uninstall it. Even though I tried the menus I can still reach sites like GitLab and GitHub. If I go to [howsmyssl.com](https://www.howsmyssl.com/) on my Vita it says I'm using TLS 1.2. Weird, although I guess it's a good think I'm supporting TLS 1.2...

So, the VPK I've currently installed on my PS Vita is [this one](https://github.com/blastrock/pkgj/files/3908697/pkgj.zip). I can't tell for sure but since it's working and since I made my previous post back in September I'm pretty sure this one is built from the mentioned Docker image.

Good luck!

**Comment by TheRadziu** [comment]
itls only replaces webkit's module, which applies to web browser only.
TL;DR webkit=/=inapp calls.

**Comment by subworx** [comment]
> @subworx Hey!
> 
> What error message do you get? I'm asking because it's been a while and when i picked up my Vita to respond to you I couldn't refresh the list, got error `0x80436007`.
> 
> This is not the original error I had with TLS so I did some research and found [this Reddit post](https://www.reddit.com/r/VitaPiracy/comments/e18tq0/psa_temp_solution_for_pkgj_issue). What that post described made it work again.
> 
> I'm not sure this will help you, morer a note to self, but before I found that fix I was trying to confirm PGKj worked without iTLS but I couldn't uninstall it. Even though I tried the menus I can still reach sites like GitLab and GitHub. If I go to [howsmyssl.com](https://www.howsmyssl.com/) on my Vita it says I'm using TLS 1.2. Weird, although I guess it's a good think I'm supporting TLS 1.2...
> 
> So, the VPK I've currently installed on my PS Vita is [this one](https://github.com/blastrock/pkgj/files/3908697/pkgj.zip). I can't tell for sure but since it's working and since I made my previous post back in September I'm pretty sure this one is built from the mentioned Docker image.
> 
> Good luck!

Thank you so much, the post contains what I wanted to setup myself after the past weekend: a HTTPS->HTTP downgrade proxy since I couldn't find one readily setup.

The error I receive is "can't get list: HttpError: sceHttpSendRequest failed: 0x80431075 HTTPS URLs may not be supported on your device".
After changing to the HTTP link, refreshing works, but downloading any Compat pack fails with "Download failure: HttpError: bad http status: 404".

Oh well, gonna read through that Reddit threat.

Thank you very much again :)

Kind regards,
sub

**Comment by TheRadziu** [comment]
default comp pack url has been fixed in 0.52, and since iTLS-Enso now supports in-app connections, its mandatory to use on firmwares 3.60-3.65. 

Now if user tries to refresh on said firmwares, error message will tell him to install iTLS-Enso in order to fix that issue.

**Comment by Zeitwaechter** [comment]
Thank you very much for the heads up!

I will try that out now that I know about the update :+1:
